### PR TITLE
WIP: Cleanup the ELF using size_t and ut64.max instead of 0 ##bin

### PR DESCRIPTION
**Pull request by trufae**
This PR is extracted from the good old wip elf-reloc fixes, but just taking the cleanup part. no reloc stuff is touched here

**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

...

**Test plan**

Review and wait for the ci

**Closing issues**

it's starting to use UT64_MAX in more places because 0 is an actual valid offset sometimes

### CODE REVIEW

File: libr/bin/format/elf/elf.c [hunk 1]
```diff
@@ -1718,6 +1702,7 @@ ut64 Elf_(r_bin_elf_get_boffset)(ELFOBJ *bin) {
 	}
 
 	for (i = 0; i < bin->ehdr.e_phnum; i++) {
+		size_t i;
```
Comment (trufae) [inline libr/bin/format/elf/elf.c:1705]: wrong scope :D

File: libr/bin/p/bin_elf.inc [hunk 2]
```diff
@@ -860,39 +857,38 @@ static void _patch_reloc(ut16 e_machine, RIOBind *iob, RBinElfReloc *rel, ut64 S
...
 	if (!obj) {
 	   	return NULL;
 	}
-	bin = obj->bin_obj;
+	struct Elf_(r_bin_elf_obj_t) *bin = obj->bin_obj;
 	if (bin->ehdr.e_type != ET_REL) {
 		return NULL;
 	}
+	size = bin->g_reloc_num * 4;
+	if (size == 0) {
```
Comment (ret2libc) [inline libr/bin/p/bin_elf.inc:882]: use OVF_MULCHK here as well.

File: libr/bin/p/bin_elf.inc [hunk 3]
```diff
@@ -669,51 +672,45 @@ static RBinReloc *reloc_convert(struct Elf_(r_bin_elf_obj_t) *bin, RBinElfReloc
...
 		if (!(relocs = Elf_(r_bin_elf_get_relocs) (bin))) {
 			return ret;
 		}
+		size_t i;
 		for (i = 0; !relocs[i].last; i++) {
 			if (!(ptr = reloc_convert (bin, &relocs[i], got_addr))) {
 				continue;
 			}
-			r_list_append (ret, ptr);
+			if (ptr->paddr) {
```
Comment (ret2libc) [inline libr/bin/p/bin_elf.inc:711]: why checking for paddr == 0?

File: libr/bin/p/bin_elf.inc [hunk 4]
```diff
@@ -569,12 +569,15 @@ static RList* libs(RBinFile *bf) {
 
 static RBinReloc *reloc_convert(struct Elf_(r_bin_elf_obj_t) *bin, RBinElfReloc *rel, ut64 GOT) {
 	r_return_val_if_fail (bin && rel, NULL);
+	if (GOT == UT64_MAX) {
```
Comment (ret2libc) [inline libr/bin/p/bin_elf.inc:572]: 1) why UT64_MAX should be here? If it is, probably the code that puts it there should be fixed
2) why if got == ut64_max you transform it into got = 0?

Comment (trufae) [reply libr/bin/p/bin_elf.inc:572]: ignore this pr the ut64 change is in another pr, ill rebase and cleanup when the others get merged

**Comment by XVilka** [comment]
Should be this one closed?

**Comment by trufae** [comment]
This is now rebased and massaged int master

**Pull request closed without merging**